drivers/mtd/gd55.c

```diff
@@ -0,0 +1,2065 @@
+ *   Write a number of blocks of data.
+ *
+ * Input Parameters:
+ *   dev        - a reference to the device structure
+ *   startblock - start block of the memory to write
+ *   nblocks    - nblocks to write
+ *   buf        - pointer to the buffer with the data to write
+ *
+ * Returned Value:
+ *   Size of the data written
+ *
+ ****************************************************************************/
+
+ssize_t gd55_bwrite(FAR struct mtd_dev_s *dev, off_t startblock,
+                    size_t nblocks, FAR const uint8_t *buf)
+{
+  FAR struct gd55_dev_s *priv = (FAR struct gd55_dev_s *)dev;
+  int                   ret;
+
+  finfo("startblock: %08lx nblocks: %d\n", (long)startblock, (int)nblocks);
+
+  /* Lock the QuadSPI bus and write all of the pages to FLASH */
+
+  gd55_lock(priv);
+
+#if defined(CONFIG_MTD_GD55_SECTOR512)
+  ret = gd55_write_cache(priv, buf, startblock, nblocks);
+  if (ret < 0)
+    {
+      ferr("ERROR: gd55_write_cache failed: %d\n", ret);
+    }
+
+#else
+  ret = gd55_write_page(priv, buf, startblock << GD55_PAGE_SHIFT,
+                          nblocks << GD55_PAGE_SHIFT);
+  if (ret < 0)
+    {
+      ferr("ERROR: gd55_write_page failed: %d\n", ret);
+    }
+#endif
+
+  gd55_unlock(priv);
+
+  return ret < 0 ? ret : nblocks;
+}
+
+/****************************************************************************
+ * Name: gd55_read
+ *
+ * Description:
+ *   Read a number of bytes of data.
+ *
+ * Input Parameters:
+ *   dev        - a reference to the device structure
+ *   offset     - starting address of the memory to read
+ *   nbytes     - nbytes to read
+ *   buf        - pointer to the buffer to store the read data
+ *
+ * Returned Value:
+ *   Size of the data read
+ *
+ ****************************************************************************/
+
+ssize_t gd55_read(FAR struct mtd_dev_s *dev, off_t offset, size_t nbytes,
+                  FAR uint8_t *buffer)
+{
+  int                   ret;
+  FAR struct gd55_dev_s *priv = (FAR struct gd55_dev_s *)dev;
+
+  finfo("offset: %08lx nbytes: %d\n", (long)offset, (int)nbytes);
+
+  /* Lock the QuadSPI bus and select this FLASH part */
+
+  gd55_lock(priv);
+  ret = gd55_read_bytes(priv, buffer, offset, nbytes);
+  gd55_unlock(priv);
+
+  if (ret < 0)
+    {
+      ferr("ERROR: gd55_read_bytes returned: %d\n", ret);
+      return (ssize_t)ret;
+    }
+
+  finfo("return nbytes: %d\n", (int)nbytes);
+  return (ssize_t)nbytes;
+}
+
+/****************************************************************************
+ * Name: gd55_ioctl
+ *
+ * Description:
+ *   IOCTLS relating to the GD55 mtd device
+ *
+ * Input Parameters:
+ *   dev        - a reference to the device structure
+ *   cmd        - ioctl command
+ *   arg        - ioctl argument
+ *
+ * Returned Value:
+ *   Success (OK) or fail (negated error code)
+ ****************************************************************************/
+
+int gd55_ioctl(FAR struct mtd_dev_s *dev, int cmd, unsigned long arg)
```

Comment:
add static to all private function.

Reply:
As above - declarations are inconsistent with definitions and although wrong nothing picked it up. Sigh!

Reply:
As above - done. Sorry!